### ISSUE TITLE
py3: Error when password has ' and is in filename

### ISSUE THREAD

**Issue by puzzledsab** [sabnzbd/sabnzbd#1300]
At least I suspect it's the '. Tested using your py3 branch, Python 3.7 and Windows 10. The password is correct. I have changed the letters but kept the form of the nzb name and password. I have tested another passworded nzb which had ' in the filename but not in the password, and that did not give this error. The error appeares every time I try the specific nzb.

2019-06-14 23:44:35,509::INFO::[misc:697] Found a password that was set by the user:  C-supersecr'et
2019-06-14 23:44:35,510::INFO::[assembler:304] Error during inspection of RAR-file \\?\d:\incomplete\Filename 1989{{C-supersecr'et}}\fileset.part09.rar
2019-06-14 23:44:35,511::DEBUG::[assembler:305] Traceback:
Traceback (most recent call last):
  File "C:\git\sabnzbd\sabnzbd\assembler.py", line 243, in check_encrypted_and_unwanted_files
    passwords = get_all_passwords(nzo)
  File "C:\git\sabnzbd\sabnzbd\misc.py", line 705, in get_all_passwords
    meta_passwords.append(pw)
AttributeError: 'str' object has no attribute 'append'

Another thing, the password is no longer removed from the file name in the web admin or the directory name.

**Comment by sanderjo** [comment]
python3 only, or also with python2?

**Comment by puzzledsab** [comment]
I can't test using Python 2 at the moment so I don't know.

**Comment by sanderjo** [comment]
I renamed a NZB with a passworded rar to `reftestnzb-8ca204ccaf4089{{C-supersecr'et}}.nzb`

On py2 I don't get a traceback

With py3, I get something else than you

```
019-06-18 20:12:36,600::INFO::[misc:697] Found a password that was set by the user:  C-supersecr'et
2019-06-18 20:12:36,600::INFO::[assembler:258] Trying password "C-supersecr'et" on job "reftestnzb-8ca204ccaf4089{{C-supersecr'et}}"
2019-06-18 20:12:36,650::INFO::[assembler:304] Error during inspection of RAR-file /home/sander/Downloads/incomplete/reftestnzb-8ca204ccaf4089{{C-supersecr'et}}/my_passworded_post.part01.rar
2019-06-18 20:12:36,650::DEBUG::[assembler:305] Traceback: 
Traceback (most recent call last):
  File "/home/sander/git/blabla-sab/sabnzbd/assembler.py", line 266, in check_encrypted_and_unwanted_files
    zf.testrar()
  File "/home/sander/git/blabla-sab/sabnzbd/utils/rarfile.py", line 874, in testrar
    check_returncode(p, output)
  File "/home/sander/git/blabla-sab/sabnzbd/utils/rarfile.py", line 2992, in check_returncode
    raise exc(msg)
sabnzbd.utils.rarfile.RarWrongPassword: None [11]: b"\nThe specified password is incorrect.\nCannot find volume /home/sander/Downloads/incomplete/reftestnzb-8ca204ccaf4089{{C-supersecr'et}}/my_passworded_post.part02.rar"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sander/git/blabla-sab/sabnzbd/assembler.py", line 275, in check_encrypted_and_unwanted_files
    if 'need to start extraction from a previous volume' in e:
TypeError: argument of type 'RarWrongPassword' is not iterable
2019-06-18 20:12:36,652::DEBUG::[directunpacker:130] DirectUnpack queued my_passworded_post.part01.rar for my_passworded_post
```

**Comment by Safihre** [comment]
Closed as part of 815703181a453d205b1bbbbb2cddfb5eaadafbce
